src/server/api/url_parser.cpp

```diff
@@ -26,15 +26,15 @@ struct URLParser final : qi::grammar<Iterator, Into>
     {
         using boost::spirit::repository::qi::iter_pos;
 
-        alpha_numeral = qi::char_("a-zA-Z0-9");
+        identifier = qi::char_("a-zA-Z0-9_..--~:");
         percent_encoding =
             qi::char_('%') > qi::uint_parser<unsigned char, 16, 2, 2>()[qi::_val = qi::_1];
-        polyline_chars = qi::char_("a-zA-Z0-9_.--[]{}@?|\\~`^") | percent_encoding;
-        all_chars = polyline_chars | qi::char_("=,;:&().");
+        polyline_chars = qi::char_("a-zA-Z0-9_..--[]{}@?|\\~`^") | percent_encoding;
+        all_chars = polyline_chars | qi::char_("=,;:&()..");
```

Comment:
Shouldn't be needed as above.

Reply:
```suggestion
        all_chars = polyline_chars | qi::char_("=,;:&().-");
```
With `-` removed from `polyline_chars`, it needs to be added here.
For example, the following test breaks otherwise:

https://github.com/Project-OSRM/osrm-backend/blob/dca35dcc8676df6fa93f491a7b157136918173d5/features/testbot/status.feature#L55